### PR TITLE
Introducing UsageDetails and Aggregated Cost by Managment Group Scope. Also updated some existing examples to include missing billingperiod in billingPeriodId Property.

### PR DESCRIPTION
**Pull request by parisa-naeimi**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by parisa-naeimi** [comment]
@marstr Hello Martin, Please take a look at this PR and let me know if I need to do something here. Also this PR needs ARMReview. Thank you

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-06-30/consumption.json [hunk 1]
```diff
@@ -2327,7 +2327,259 @@
...
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/providers/Microsoft.Management/managementGroups/{managementGroupId}/providers/Microsoft.Billing/billingPeriods/{billingPeriodName}/providers/Microsoft.Consumption/usageDetails": {
+      "get": {
+        "tags": [
+          "UsageDetails"
+        ],
+        "operationId": "UsageDetails_ListByForBillingPeriodByManagementGroup",
```
Comment (marstr) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-06-30/consumption.json]: `ListByForBillingPeriodByManagementGroup` seems like a typo to me. Was it supposed to be `ListByBillingPeriodByManagementGroup`?

**Review by marstr — COMMENTED** [review]
I have a few comments that should keep your operation groups consistent through-out this swagger.

 In general the pattern here does feel kind of backwards here, shouldn't there be an operation group per scope that can have consumption data viewed? Then have UsageDetails and AggregatedCost operations from there? It's not actionable ATM, because it would cause breaking changes without just cause. However, it may be food for thought in future API Versions.

**Review by marstr — CHANGES_REQUESTED** [review]
Please fix the individual comments noted above.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-06-30/consumption.json [hunk 2]
```diff
@@ -2327,7 +2327,259 @@
...
         }
       }
-    }
+    },
+	"/providers/Microsoft.Management/managementGroups/{managementGroupId}/providers/Microsoft.Consumption/usageDetails": {
+      "get": {
+        "tags": [
+          "UsageDetails"
+        ],
+        "operationId": "UsageDetailsByManagementGroup_List",
```
[the remark this answers is not in the document]
Comment (parisa-naeimi) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-06-30/consumption.json]: Sure, will apply your comment

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-06-30/consumption.json [hunk 3]
```diff
@@ -2327,7 +2327,259 @@
...
+          "nextLinkName": "nextLink"
+        }
+      }
+	},
+	"/providers/Microsoft.Management/managementGroups/{managementGroupId}/providers/Microsoft.Consumption/aggregatedcost": {
+      "get": {
+        "tags": [
+          "AggregatedCost"
+        ],
+        "operationId": "AggregatedcostByManagementGroup",
```
[the remark this answers is not in the document]
Comment (parisa-naeimi) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-06-30/consumption.json]: Sure, will apply it

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-06-30/consumption.json (hunk 1 above)
Comment (parisa-naeimi) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-06-30/consumption.json]: Yeah, It is a typo Good Catch  👍

**Comment by parisa-naeimi** [comment]
@marstr I addressed all of your comments. Thanks again.

**Review by marstr — CHANGES_REQUESTED** [review]
Look like the linter has a few good suggestions that should be fixed up. Both the missing `x-ms-parameter-location` on the global parameter and making sure that Get is added to those OperationIDs we were looking at before. Sorry, I should have caught the missing Get or List earlier!

**Comment by parisa-naeimi** [comment]
@marstr No worries, I will fix those ones as well. Is it the reason it is failing? cause these ones seem to me as a warning

**Comment by marstr** [comment]
> Is it the reason it is failing?

Adding new warnings or errors causes the job to fail, but we only block on it if the new warnings or errors are needless. This time we should fix them up, even though they're just warnings :)

**Comment by parisa-naeimi** [comment]
Can you give me an example for using location parameter? what do you expect to see there?

**Comment by marstr** [comment]
Yeah, sorry it looks like the documentation the linter links to is missing this rule for some reason. :/ I'll look into that.

Anyway, here's an example of the SQL team using it:
https://github.com/Azure/azure-rest-api-specs/blob/0b33b0b52d88a871bc92a1e6e8fa8d47a971538a/specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/cancelPoolOperations.json#L259

It needs to be applied on any document-scoped parameter definition so that AutoRest knows whether or not to ask for them as part of a client constructor or in the methods that are generated. You can read more about it here: https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-parameter-location

**Comment by parisa-naeimi** [comment]
@marstr  Thank you, I fixed those and sent another iteration

**Review by marstr — APPROVED** [review]
LGTM, now we'll just need to wait for the ARM team.

**Comment by ravbhatnagar** [comment]
Looks good

**Pull request merged**